src/Expander.hs

```diff
@@ -1191,17 +1189,19 @@ runTask (tid, localData, task) = withLocal localData $ do
         Just _v -> do
           bind b $ EVarMacro x
           forkExpandSyntax (ExprDest t dest) stx
-    ExpandDecl dest sc stx ph ->
-      expandOneDeclaration sc dest stx ph
-    ExpandMoreDecls dest sc stx waitingOn -> do
-      readyYet <- view (expanderDeclPhases . at waitingOn) <$> getState
+    ExpandDeclForms dest earlierScopeSet waitingOn vp stx -> do
+      -- The scopes in waitingOn (from the preceding declaration) haven't yet
+      -- been added to stx, while those in earlierScopeSet (from the
+      -- declarations before that) have. The scopes from both, plus the scopes
+      -- introduced by the declaration forms in stx, must all eventually be
+      -- placed in vp so that the code which isn't part of this declaration
+      -- group can have access to those declarations.
```

Comment:
Here is one of the comments which explains how I'm accumulating the scopes now.

Reply:
That's great!  What is `vp` an abbreviation for?

Reply:
"validity pointer". Maybe "dvp" would be better?

Reply:
What about `outScopes` or `scopeOutDest` or something else that at includes some recognizable words?

Reply:
ok, I've renamed `DeclValidityPtr` to `DeclOutputScopesPtr` and `vp` to `outScopesDest`.